examples/Python/PET/asarray.py

```diff
@@ -0,0 +1,145 @@
+scheme = pet.AcquisitionData.get_storage_scheme()
+if scheme != storage:
+    print('default storage scheme is %s' % repr(scheme))
+    print('setting storage scheme to %s' % repr(storage))
+    pet.AcquisitionData.set_storage_scheme(storage)
+else:
+    print('using default storage scheme %s' % repr(scheme))
+
+
+def test_img_asarray(img_data):
+    try:
+        img_asarray = img_data.asarray() # view
+        print('img_data.asarray() ok')
+    except Exception:
+        print('data not contiguous, making a copy...')
+        img_data = img_data.copy()
+        img_asarray = img_data.asarray()
+    img_as_array = img_data.as_array() # deepcopy
+    #print(numpy.linalg.norm(img_asarray))
+    #print(numpy.linalg.norm(img_as_array))
+    diff = img_asarray - img_as_array
+    print('norm of img_data.asarray() - img_data.as_array(): %f' % numpy.linalg.norm(diff))
+    img_asarray += 1 # img_data changed
+    img_array = img_data.as_array()
+    #print(numpy.linalg.norm(img_asarray))
+    #print(numpy.linalg.norm(img_array))
+    diff = img_asarray - img_array
+    print('norm of img_data.asarray() - img_data.as_array(): %f' % numpy.linalg.norm(diff))
+
+
+def main():
+    engine_version = pet.get_engine_version_string()
+    print('Using %s version %s as the reconstruction engine' % (engine, engine_version))
+    print('%s doc path: %s' % (engine, pet.get_engine_doc_dir()))
+    print('%s examples path: %s' % (engine, pet.get_engine_examples_dir()))
+
+    # direct all engine's messages to files
+    _ = pet.MessageRedirector('info.txt', 'warn.txt', 'errr.txt')
+
+    # PET acquisition data to be read from this file
+    raw_data_file = existing_filepath(data_path, data_file)
+    print('raw data: %s' % raw_data_file)
+    acq_data = pet.AcquisitionData(raw_data_file)
+
+    # copy the acquisition data into a Python array
+    dim = acq_data.dimensions()
+    print('data dimensions: %d x %d x %d x %d' % dim)
+    acq_array = acq_data.as_array()
+
+    print('\n== Testing asarray() method of pet.AcquisitionData...')
+
+#    acq_asarray = numpy.asarray(acq_data)
+    acq_asarray = acq_data.asarray() # same as above
+    diff = acq_array - acq_asarray
+    print('norm of acq_data.as_array() - acq_data.asarray(): %f' % numpy.linalg.norm(diff))
+
+    print('\n== Testing asarray() method of pet.ImageData...')
+
+    print('- testing image from file:')
+    img_data = pet.ImageData(existing_filepath(data_path, 'mMR/mu_map.hv')) # contiguous
+    test_img_asarray(img_data)
+
+    print('- testing image returned by AcquisitionData.create_uniform_image():')
+    img_data = acq_data.create_uniform_image(5)
+    test_img_asarray(img_data)
+
+    print('- testing image constructed from acquisition data:')
+    img_data = pet.ImageData(acq_data)
+    test_img_asarray(img_data)
+
+    _ = img_data.asarray() # view dummy to trigger contiguity error message
+
+    print('ok') # should not be printed (error in the previous line)
```

Comment:
this is not a way to test if this error gets thrown.

Reply:
Also, will it now actually throw, or just return a copy anyway?

Reply:
now use `try`-`catch` instead